allennlp/data/token_indexers/pretrained_transformer_indexer.py

```diff
@@ -101,6 +101,43 @@ def tokens_to_indices(self, tokens: List[Token], vocabulary: Vocabulary) -> Inde
 
         return self._postprocess_output(output)
 
+    @overrides
+    def indices_to_tokens(
+        self, indexed_tokens: IndexedTokenList, vocabulary: Vocabulary
+    ) -> List[Token]:
+        token_ids = indexed_tokens["token_ids"]
+        type_ids = indexed_tokens["type_ids"] if "type_ids" in indexed_tokens else None
+
+        token_ids = self._preprocess_segmented_tokens(token_ids)
+
+        return [
+            Token(
+                text=vocabulary.get_token_from_index(token_ids[i], self._namespace),
+                text_id=token_ids[i],
+                type_id=type_ids[i] if type_ids is not None else None,
+            )
+            for i in range(len(token_ids))
+        ]
+
+    def _preprocess_segmented_tokens(self, tokens: List[Token]):
```

Comment:
I agree that it's not a great solution, I rewrote it to simply remove all special tokens and add special tokens back in afterwards.

I think that in reality we won't ever get such sequences in `indices_to_tokens`. To my understanding max length is there to fold sequences that are too long for the given transformer into shorter segments. I think currently the only use case of max length is for the `PretrainedTransformerEmbedder`, which takes care of unfolding itself (besides that, those are embeddings and not ids anyways).

I am not sure yet how we should deal with models that BART that generate variable length output. What if we feed in a very long article that is split up into several segments and then text is generated for each segment? Each segment is missing context from other segments, so we can't expect the text generated from each segment to make sense, especially not if we would simply concatenate them together.

Anyways, the reason I included that code is to make `indices_to_tokens` a proper inverse
of `tokens_to_indices`. Otherwise, if you use `tokens_to_indices` with `max_length` and then use `indices_to_tokens`, you will get different output due to the special tokens.

Reply:
Trying to make them perfect inverses of each other is a good idea, but I think it's too complex in this case. Let's remove that part of the code, and document that in the presence of `_max_length`, this is not a perfect inverse.